src/lib/chat/matrix-client.ts

```diff
@@ -412,16 +413,12 @@ export class MatrixClient implements IChatClient {
     return '';
   }
 
-  private getLastMessageEvent(room: Room) {
-    const timelineEvents = room.getLiveTimeline().getEvents();
-
-    for (let i = timelineEvents.length - 1; i >= 0; i--) {
-      if (timelineEvents[i].getType() === EventType.RoomMessage) {
-        return timelineEvents[i];
-      }
-    }
-
-    return null;
+  private getAllMessagesFromRoom(room: Room) {
+    const timeline = room.getLiveTimeline().getEvents();
+    const messages = timeline
+      .filter((event) => event.getType() === EventType.RoomMessage)
+      .map(this.mapMatrixEventToMessage);
+    return messages;
```

Comment:
Will this return "all" messages, or is there a default limit? (like 30 messages)

Reply:
Yes I believe this will filter all events equal to `m.room.message`, map the event to message, then return all messages. @dalefukami / @ratik21 , should we add a limit? Would a limit cause any unexpected behaviour?

Reply:
If there's a limit then we should add a limit...as long as the limit applies to the most recent messages (and not just getting the first 30. lol)

Reply:
I found that **_if_** we use `createMessagesRequest` there is a limit prop we can use. This would change the set up for getting messages a little, so i will merge this PR and revisit 👍